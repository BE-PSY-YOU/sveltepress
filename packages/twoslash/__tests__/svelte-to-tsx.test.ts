import { readFileSync } from 'node:fs'
import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'
import { svelte2tsx } from 'svelte2tsx'

const svelteFilePath = resolve(import.meta.dirname, 'test.svelte')
const svelteCode = readFileSync(svelteFilePath, 'utf-8')

describe('svelte to tsx', () => {
  it('tsx code', async () => {
    expect(svelte2tsx(svelteCode).code).toMatchSnapshot()
  })
})
