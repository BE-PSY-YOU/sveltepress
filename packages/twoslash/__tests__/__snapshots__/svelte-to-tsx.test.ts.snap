// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`svelte to tsx > tsx code 1`] = `
"///<reference types="svelte" />
;
  export const FOO = 'BAR'
;;


import { onMount } from 'svelte'
import FloatingWrapper from '@sveltepress/twoslash/FloatingWrapper.svelte'
function render() {

  
  
  

  let count = 0

  onMount(() => {
    console.log('mounted')
  })
;
async () => {



 { svelteHTML.createElement("button", {  "on:click":() => count++,});
    count;
 }

 { const $$_repparWgnitaolF0C = __sveltets_2_ensureComponent(FloatingWrapper); const $$_repparWgnitaolF0 = new $$_repparWgnitaolF0C({ target: __sveltets_2_any(), props: { "placement":\`right\`,}});
   { svelteHTML.createElement("div", { "class":\`text-xl b-1 b-solid b-blue rounded py-10 px-4\`,});
    
   }

   {const {/*立ignore_start立*/$$_$$/*立ignore_end立*/,} = $$_repparWgnitaolF0.$$slot_def["floating-content"];$$_$$;{ svelteHTML.createElement("div", {  "class":\`bg-white dark:bg-dark b-solid b-1 b-red rounded p-4\`,});
     
   }}
 FloatingWrapper}};
return { props: /** @type {Record<string, never>} */ ({}), slots: {}, events: {} }}

export default class extends __sveltets_2_createSvelte2TsxComponent(__sveltets_2_partial(__sveltets_2_with_any_event(render()))) {
}"
`;
